core/src/css/ionic/core.ionic.scss

```diff
@@ -36,45 +36,76 @@ body.backdrop-no-scroll {
 
 // Modal - Card Style
 // --------------------------------------------------
+html.ionic ion-modal ion-header {
+  box-shadow: none;
+}
+
 /**
- * Card style modal needs additional padding on the
+ * Modals need additional padding on the
  * top of the header. We accomplish this by targeting
  * the first toolbar in the header.
  * Footer also needs this. We do not adjust the bottom
  * padding though because of the safe area.
  */
-html.ios ion-modal.modal-card ion-header ion-toolbar:first-of-type,
-html.ios ion-modal.modal-sheet ion-header ion-toolbar:first-of-type,
-html.ios ion-modal ion-footer ion-toolbar:first-of-type {
-  padding-top: globals.$ionic-space-150;
+html.ionic ion-modal.modal-card ion-header ion-toolbar:first-of-type,
+html.ionic ion-modal.modal-sheet ion-header ion-toolbar:first-of-type,
+html.ionic ion-modal ion-footer ion-toolbar:first-of-type {
+  padding-top: globals.$ionic-space-400;
 }
 
 /**
-* Card style modal needs additional padding on the
+* Modals need additional padding on the
 * bottom of the header. We accomplish this by targeting
 * the last toolbar in the header.
 */
-html.ios ion-modal.modal-card ion-header ion-toolbar:last-of-type,
-html.ios ion-modal.modal-sheet ion-header ion-toolbar:last-of-type {
-  padding-bottom: globals.$ionic-space-150;
+html.ionic ion-modal.modal-card ion-header ion-toolbar:last-of-type,
+html.ionic ion-modal.modal-sheet ion-header ion-toolbar:last-of-type,
+html.ionic ion-modal ion-footer ion-toolbar:last-of-type {
+  padding-bottom: globals.$ionic-space-400;
+}
+
+/**
+* Add additional padding to the left and right of toolbars while accounting
+* for the safe area insets.
+*/
+html.ionic ion-modal ion-toolbar {
+  --padding-start: calc(var(--ion-safe-area-right) + #{globals.$ionic-space-400});
+  --padding-end: calc(var(--ion-safe-area-left) + #{globals.$ionic-space-400});
+}
+
+/**
+* Center an `ion-title` within a modal header. This works around
+* limitations with `:host-context()` not being supported in all browsers.
+*/
+html.ionic ion-modal ion-header ion-toolbar ion-title {
+  @include globals.position(0, null, null, 0);
+
+  position: absolute;
+
+  width: 100%;
+  height: 100%;
+
+  transform: translateZ(0);
+
+  text-align: center;
```

Comment:
I had to override the styles for `ion-title` because the ionic theme styles for that component use `:host-context()` which is supported in Safari and Firefox. I can make this change in the title styles themselves, but it does result in some screenshots changing and I wasn't exactly sure what the correct styles were for that component